libs/cri-response-service/src/main/java/uk/gov/di/ipv/core/library/service/CriResponseService.java

```diff
@@ -62,15 +69,35 @@ public void persistCriResponse(
         dataStore.create(criResponseItem, ConfigurationVariable.CRI_RESPONSE_TTL);
     }
 
-    public CriResponseItem getFaceToFaceRequest(String userId) {
-        return dataStore.getItem(userId, "f2f");
-    }
-
     public void deleteCriResponseItem(String userId, Cri credentialIssuer) {
         dataStore.delete(userId, credentialIssuer.getId());
     }
 
     public void updateCriResponseItem(CriResponseItem responseItem) {
         dataStore.update(responseItem);
     }
+
+    public AsyncCriStatus getAsyncResponseStatus(
+            String userId,
+            Function<Cri, Boolean> hasVc,
+            Function<Cri, Long> getVcIat,
+            boolean isPendingEvcsIdentity) {
+        final var criResponseItem =
+                dataStore.getItems(userId).stream()
+                        .filter(item -> ASYNC_CRIS.contains(Cri.fromId(item.getCredentialIssuer())))
+                        .findFirst()
+                        .orElse(null);
+        if (Objects.isNull(criResponseItem)) {
+            return new AsyncCriStatus(null, null, null, false, false);
+        }
+
+        var cri = Cri.fromId(criResponseItem.getCredentialIssuer());
+        final boolean hasNoVc = !hasVc.apply(cri);
+        final boolean isComplete =
+                hasVc.apply(cri)
+                        && (!configService.enabled(EVCS_READ_ENABLED) || isPendingEvcsIdentity);
```

Comment:
Assume we don't need to save/ consider other VCs from before strategic app, i.e. won't think about EVCS for dcmaw async.